### PR TITLE
Implemented ILogger interface (#6)

### PR DESCRIPTION
**Pull request by dhindrik**

### CODE REVIEW

File: TinyInsights.TestApp/MainPage.xaml.cs [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
+    }
+}
+
     private async void PageViewButton_OnClicked(object? sender, EventArgs e)
     {
+        if(UseILogger)
+        {
+            logger.LogTrace("MainView");
+            return;
+        }
```
Comment (Kebechet) [inline TinyInsights.TestApp/MainPage.xaml.cs:40]: are these checks really necessary ? 
Wouldnt be better to implement one signle check into `InsightsExtension.cs` ?

File: TinyInsights/InsightsExtension.cs [hunk 2]
```diff
@@ -28,4 +30,22 @@ public static MauiAppBuilder UseTinyInsights(this MauiAppBuilder appBuilder, str
...
+        appBuilder.Services.AddSingleton<ILogger>((services) =>
+        {
+            var provider = new ApplicationInsightsProvider(applicationInsightsConnectionString);
+            
+            configureProvider?.Invoke(provider);
+
+            return provider;
+        });
+
+        appBuilder.Services.AddTransient<InsightsMessageHandler>();
```
Comment (Kebechet) [inline TinyInsights/InsightsExtension.cs:46]: Should the message handler be really transient ?

File: TinyInsights/ApplicationInsightsProvider.cs [hunk 3]
```diff
@@ -309,4 +310,61 @@ public Task TrackDependencyAsync(string dependencyType, string dependencyName, s
...
+    public async void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+    {
+        if(!IsEnabled(logLevel))
+        {
+            return;
+        }
+
+        var logTask = logLevel switch {
+            LogLevel.Trace => TrackPageViewAsync(GetEventName(eventId), GetLoggerData(logLevel, eventId, state, exception, formatter)),
+            LogLevel.Debug => Task.CompletedTask,
```
Comment (Kebechet) [inline TinyInsights/ApplicationInsightsProvider.cs:324]: is the Debug option correct ?

File: TinyInsights.TestApp/MainPage.xaml.cs (hunk 1 above)
Comment (dhindrik) [reply TinyInsights.TestApp/MainPage.xaml.cs:40]: No it is just for test purpose. So I can test both ways to log in an easy way.

File: TinyInsights/InsightsExtension.cs (hunk 2 above)
Comment (dhindrik) [reply TinyInsights/InsightsExtension.cs:46]: My idea is that the lifetime then will be handled by the HttpClient. But if you have opinions about that I will listen.

File: TinyInsights/ApplicationInsightsProvider.cs (hunk 3 above)
Comment (dhindrik) [reply TinyInsights/ApplicationInsightsProvider.cs:324]: Do you have any suggestion how I should log Debug data?

Comment (dhindrik) [reply TinyInsights/ApplicationInsightsProvider.cs]: I now use Debug.WriteLine to output data there. What do you think about that?

**Pull request merged**